### PR TITLE
Added scale down flag to cluster autoscaler

### LINKED ISSUE #1030: Allow to configure scale-down-utilization-threshold for cluster-autoscaler
**What would you like to be added**:
By default the scale down of a node happens when the utilisation of CPU/RAM goes below 50% [0]. It would be great to expose the `scale-down-utilization-threshold` flag as a shoot configuration option in the spec.

```
    cluster-autoscaler:
      enabled: true
      scaleDownUtilizationThreshold: 0.6
```

**Why is this needed**:
If a customer knows exactly how his workload looks like, he might be able to run his shoot cluster more cost efficient.

/cc @usherfu

[0] https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/main.go#L96

### PR DESCRIPTION
**Pull request by afritzler**
**What this PR does / why we need it**:
By default the scale down of a node happens when the utilisation of CPU/RAM goes below 50%. It would be great to expose the scale-down-utilization-threshold flag as a shoot configuration option in the spec.

```
    cluster-autoscaler:
      enabled: true
      scaleDownUtilizationThreshold: 0.6
```

**Which issue(s) this PR fixes**:
Fixes #1030 #546

**Release note**:
```improvement operator
It is now possible to configure several command line flags that influence the cluster autoscaler behaviour in the `.spec.kubernetes.clusterAutoscaler` configuration for shoot clusters.
```

### CODE REVIEW

File: pkg/apis/garden/v1beta1/types.go [hunk 1]
```diff
@@ -1166,6 +1166,9 @@ type KubernetesDashboard struct {
 // ClusterAutoscaler describes configuration values for the cluster-autoscaler addon.
 type ClusterAutoscaler struct {
 	Addon `json:",inline"`
+	// ScaleDownUtilizationThreshold defines the threshold in % under which a node is being removed
+	// +optional
+	ScaleDownUtilizationThreshold *string `json:"scale-down-utilization-threshold,omitempty"`
```
Comment (mvladev) [inline pkg/apis/garden/v1beta1/types.go]:
```suggestion
	ScaleDownUtilizationThreshold *string `json:"scaleDownUtilizationThreshold,omitempty"`
```

Comment (mvladev) [reply pkg/apis/garden/v1beta1/types.go]: This should follow K8S naming conventions - see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions

**Review by mvladev — DISMISSED** [review]

File: pkg/apis/garden/v1beta1/types.go (hunk 1 above)
Comment (afritzler) [reply pkg/apis/garden/v1beta1/types.go]: Good point!

**Comment by afritzler** [comment]
I just realized, that we are deprecating the autoscaler addon in the shoot spec. :-/ Not sure what to do about that.

**Comment by rfranzke** [comment]
There is also https://github.com/gardener/gardener/issues/546 (pretty much a duplicate of #1030).
`.spec.addons.cluster-autoscaler` is deprecated and will be removed soon, yes.
WDYT about introducing `.spec.kubernetes.clusterAutoscaler` (next to the other control plane components) and expose it there?

**Comment by afritzler** [comment]
Yes, `.spec.kubernetes.clusterAutoscaler` is probably a better place for the auto scaler configuration. Let me refactor this.

**Comment by afritzler** [comment]
Ok, I moved the flags to `spec.kubernetes.clusterAutoscaler`.  I totally missed the discussion in #546 on this topic. @rfranzke do you think we should freeze this for now or should we go ahead and expose the autoscaler configuration in the `kuberentes` part of the shoot spec?

File: pkg/apis/garden/types.go [hunk 2]
```diff
@@ -1308,6 +1308,15 @@ type Kubernetes struct {
...
 	Version string
+	// ClusterAutoscaler contains the configration flags for the Kubernetes cluster autoscaler.
+	ClusterAutoscaler *ClusterAutoscaler `json:"clusterAutoscaler,omitempty"`
+}
+
+// ClusterAutoscaler contains the configration flags for the Kubernetes cluster autoscaler.
+type ClusterAutoscaler struct {
+	// ScaleDownUtilizationThreshold defines the threshold in % under which a node is being removed
+	// +optional
+	ScaleDownUtilizationThreshold *float64
```
Comment (adracus) [inline pkg/apis/garden/types.go:1319]: What about validation for this field? Also, maybe we should use `IntOrString` for this as it's done e.g. for a deployment's `MaxUnavailable` field.

**Review by adracus — DISMISSED** [review]

File: pkg/apis/garden/types.go (hunk 2 above)
Comment (afritzler) [reply pkg/apis/garden/types.go:1319]: Well, actually this has to be a float.

Comment (adracus) [reply pkg/apis/garden/types.go:1319]: Well, you're completely right :see_no_evil: was too early for me in the morning. However, still, what about validation? Negative values don't make that much sense here.

**Review by adracus — APPROVED** [review]

**Comment by rfranzke** [comment]
@afritzler can you also add the other flags (https://github.com/gardener/gardener/issues/546) that make sense to be exposed?

**Comment by afritzler** [comment]
Sure thing. I will create a separate PR for the fine grained options.

**Pull request merged**